src/Widgets/ScreenShield.vala

```diff
@@ -0,0 +1,115 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ * SPDX-FileCopyrightText: 2025 elementary, Inc. (https://elementary.io)
+ */
+
+public class Gala.ScreenShield : Clutter.Actor {
+    public signal void user_action ();
+
+    public delegate void OnAnimationEnd ();
+    public WindowManager wm { private get; construct; }
+    public bool active { get; private set; default = false; }
+
+    private ModalProxy? modal_proxy;
+
+    public ScreenShield (WindowManager wm) {
+        Object (wm: wm);
+    }
+
+    construct {
+        reactive = true;
+        visible = false;
+        opacity = 0;
+
+#if HAS_MUTTER47
+        background_color = Cogl.Color.from_string ("black");
+#else
+        background_color = Clutter.Color.from_string ("black");
+#endif
+    }
+
+    public override void parent_set (Clutter.Actor? old_parent) {
+        clear_constraints ();
+
+        unowned var parent = get_parent ();
+        if (parent == null) {
+            return;
+        }
+
+        add_constraint (new Clutter.BindConstraint (parent, Clutter.BindCoordinate.SIZE, 0.0f));
+    }
+
+    public override bool button_press_event (Clutter.Event event) {
+        user_action ();
+        return Clutter.EVENT_STOP;
+    }
+
+    public override bool key_press_event (Clutter.Event event) {
+        user_action ();
+        return Clutter.EVENT_STOP;
+    }
+
+    public override bool motion_event (Clutter.Event event) {
```

Comment:
Can we collect all events at once in `event (Clutter.Event event)` and filter them by type instead of having three handlers doing the same thing?